src/fights.ts

```diff
@@ -1775,41 +1778,70 @@ const fireExtinguishZones = [
+    openCost: () => 0,
+    preReq: null,
   },
   {
     location: $location`The Ice Hotel`,
     monster: $monster`ice bartender`,
     item: $item`perfect ice cube`,
     dropRate: 1,
     maximize: [],
-    open: () => get("_coldAirportToday") || get("coldAirportAlways"),
+    isOpen: () => get("_coldAirportToday") || get("coldAirportAlways"),
+    openCost: () => 0,
+    preReq: null,
   },
   {
     location: $location`The Haunted Library`,
     monster: $monster`bookbat`,
     item: $item`tattered scrap of paper`,
     dropRate: 1,
     maximize: ["99 monster level 100 max"], // Bookbats need up to +100 ML to survive the polar vortices
-    open: () => have($item`[7302]Spookyraven library key`),
+    isOpen: () => have($item`[7302]Spookyraven library key`),
+    openCost: () => 0,
+    preReq: null,
+  },
+  {
+    location: $location`The Stately Pleasure Dome`,
+    monster: $monster`toothless mastiff bitch`,
+    item: $item`disintegrating spiky collar`,
+    dropRate: 1,
+    maximize: ["99 muscle 100 max"], // Ensure mastiff is at least 100 hp
+    isOpen: () => true,
+    openCost: () =>
+      !have($effect`Absinthe-Minded`) ? mallPrice($item`tiny bottle of absinthe`) : 0,
+    preReq: () => {
+      if (!have($effect`Absinthe-Minded`)) {
+        if (!have($item`tiny bottle of absinthe`)) buy(1, $item`tiny bottle of absinthe`);
+        use($item`tiny bottle of absinthe`);
+      }
+    },
   },
   {
     location: $location`Twin Peak`,
     monster: $monster`bearpig topiary animal`,
     item: $item`rusty hedge trimmers`,
     dropRate: 0.5,
     maximize: ["99 monster level 11 max"], // Topiary animals need an extra 11 HP to survive polar vortices
-    open: () => myLevel() >= 9 && get("chasmBridgeProgress") >= 30 && get("twinPeakProgress") >= 15,
+    isOpen: () =>
+      myLevel() >= 9 && get("chasmBridgeProgress") >= 30 && get("twinPeakProgress") >= 15,
+    openCost: () => 0,
+    preReq: null,
   },
 ] as fireExtinguisherZone[];
 
 let bestFireExtinguisherZoneCached: fireExtinguisherZone | undefined = undefined;
 function getBestFireExtinguisherZone(): fireExtinguisherZone | undefined {
   if (bestFireExtinguisherZoneCached !== undefined) return bestFireExtinguisherZoneCached;
-  const targets = fireExtinguishZones.filter((zone) => zone.open() && !isBanished(zone.monster));
-  bestFireExtinguisherZoneCached = targets.sort(
-    (a, b) => b.dropRate * getSaleValue(b.item) - a.dropRate * getSaleValue(a.item)
-  )[0];
+  const targets = fireExtinguishZones.filter((zone) => zone.isOpen() && !isBanished(zone.monster));
+  const vorticesAvail = Math.floor(get("_fireExtinguisherCharge") / 10);
+  bestFireExtinguisherZoneCached = targets.sort((a, b) => {
+    const value = (zone: fireExtinguisherZone): number => {
+      return zone.dropRate * getSaleValue(zone.item) * vorticesAvail - zone.openCost();
+    };
+    return value(b) - value(a);
+  })[0];
   return bestFireExtinguisherZoneCached;
```

Comment:
make sure the value is greater than 0 if we want to use it, as you pointed out in the discord.